compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -50,12 +105,63 @@ OMR::ARM64::TreeEvaluator::lsubEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
 	}
 
+// Multiply a register by a 32-bit constant
+static void mulConstant32(TR::Node *node, TR::Register *treg, TR::Register *sreg, int32_t value, TR::CodeGenerator *cg)
+   {
+   if (value == 0)
+      {
+      loadConstant32(cg, node, 0, treg);
+      }
+   else if (value == 1)
+      {
+      generateMovInstruction(cg, node, treg, sreg);
+      }
+   else if (value == -1)
+      {
+      generateNegInstruction(cg, node, treg, sreg);
+      }
+   else
+      {
+      TR::Register *tempReg = cg->allocateRegister();
+      loadConstant32(cg, node, value, tempReg);
+      generateMulInstruction(cg, node, treg, sreg, tempReg);
```

Comment:
If `tempReg` is a temporary register, you'll have to manually kill its live range here:
`cg->stopUsingRegister(tempReg)`.